src/server/game/Entities/Creature/GossipDef.cpp

```diff
@@ -85,37 +85,37 @@ void GossipMenu::AddMenuItem(uint32 menuId, uint32 menuItemId, uint32 sender, ui
     for (GossipMenuItemsContainer::const_iterator itr = bounds.first; itr != bounds.second; ++itr)
     {
         /// Find the one with the given menu item id.
-        if (itr->second.OptionIndex != menuItemId)
+        if (itr->second.TextID != menuItemId)
             continue;
 
         /// Store texts for localization.
         std::string strOptionText, strBoxText;
-        BroadcastText const* optionBroadcastText = sObjectMgr->GetBroadcastText(itr->second.OptionBroadcastTextId);
-        BroadcastText const* boxBroadcastText = sObjectMgr->GetBroadcastText(itr->second.BoxBroadcastTextId);
+        BroadCastText const* optionBroadCastText = sObjectMgr->GetBroadCastText(itr->second.OptionBroadCastTextID);
```

Comment:
it's what the changes proposed by malcrom looks like https://pastebin.com/L42KCeB6